### PR TITLE
TST: Skip Cython test for editable install

### PR DESCRIPTION
**Pull request by thalassemia**
Skip testing Cython extension for editable installs due to issues identified in https://github.com/scipy/scipy/issues/20540.

### CODE REVIEW

**Review by seberg — APPROVED** [review]
See also https://github.com/cython/cython/issues/5820 for a similar discussion.

This LGTM, I am not 100% sure if there may be a better way to detect "editable", but this also seems fine to just use.

This doesn't remove the pain that you may not be able to develop downstream cython modules if you install NumPy with an editable install.  But there isn't much to do about that...

**Comment by rgommers** [comment]
Thanks @thalassemia! I tweaked the check for editable install a bit to ensure it works with `meson-python <0.16.0`, and then used it to mark all other failing tests. This was never complete it looks like. Now `pytest numpy` passes.

File: pytest.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [pytest]
 addopts = -l
-norecursedirs = doc tools numpy/linalg/lapack_lite numpy/_core/code_generators
+norecursedirs = doc tools numpy/linalg/lapack_lite numpy/_core/code_generators numpy/_core/src/common/pythoncapi-compat
```
Comment (ngoldbaum) [inline pytest.ini:3]: oops, thanks for catching that I missed this when I added the submodule

**Review by ngoldbaum — COMMENTED** [review]
Should we maybe add a smoketest run with an editable build of numpy?

**Comment by rgommers** [comment]
> Should we maybe add a smoketest run with an editable build of numpy?

Yes we definitely should. I'd prefer to open a follow-up issue for that though, since making CI happy that time/iterations.

**Comment by ngoldbaum** [comment]
SGTM, merging. Thanks @thalassemia!

**Pull request merged**